File: ticketdesk-ui/src/services/attachmentService.js
import api from './api';

export const attachmentService = {
  uploadFile: async (file, ticketId) => {
    const formData = new FormData();
    formData.append('file', file);
    if (ticketId) formData.append('ticketId', ticketId);

    const response = await api.post('/attachments/upload', formData, {
      headers: { 'Content-Type': 'multipart/form-data' }
    });
    return response.data;
  },

  generatePresignedUrl: async (presignData) => {
    // presignData: { fileName, fileType, fileSize, ticketId }
    const response = await api.post('/attachments/presign', presignData);
    return response.data;
  },

  confirmAttachment: async (ticketId, attachmentId) => {
    const response = await api.post(`/attachments/${ticketId}/confirm?attachmentId=${attachmentId}`);
    return response.data;
  },

  getAttachmentsByTicketId: async (ticketId) => {
    const response = await api.get(`/attachments/ticket/${ticketId}`);
    return response.data;
  }
};

File: attachment-service/src/main/java/com/ticketdesk/attachmentservice/service/AttachmentService.java
package com.ticketdesk.attachmentservice.service;

import com.ticketdesk.attachmentservice.dto.request.ConfirmAttachmentRequest;
import com.ticketdesk.attachmentservice.dto.request.PresignUrlRequest;
import com.ticketdesk.attachmentservice.dto.response.AttachmentResponse;
import com.ticketdesk.attachmentservice.dto.response.PresignUrlResponse;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface AttachmentService {

    PresignUrlResponse generatePresignedUrl(PresignUrlRequest request);

    AttachmentResponse uploadFile(MultipartFile file, Long ticketId);

    AttachmentResponse confirmAttachment(Long ticketId, Long attachmentId);

    AttachmentResponse confirmAttachment(ConfirmAttachmentRequest request);

    List<AttachmentResponse> getAttachmentsByTicketId(Long ticketId);
}
